tests/manage/z_cluster/nodes/test_node_replacement_proactive.py

```diff
@@ -301,3 +302,65 @@ def test_nodereplacement_twice(self):
             assert not (
                 node_name_to_delete in str(tree_output)
             ), f"Deleted host {node_name_to_delete} still exist in ceph osd tree after node replacement"
+
+
+@tier4b
+@bugzilla("1904302")
+@ignore_leftovers
+@vsphere_platform_required
+class TestDiskReplacementNodeReplacement(ManageTest):
+    """
+    Automates osd disk replacement and new osd running node replacement scenario
+    """
+
+    @pytest.mark.polarion_id("OCS-2658")
+    def test_disk_replacement_node_replacement(
+        self, nodes, pvc_factory, pod_factory, bucket_factory, rgw_bucket_factory
+    ):
+        """
+        1. Get osds before node replacement
+        2. Get OSD running nodes
+        3. Replace disk
+        4. Get new osd and its node
+        5. Replace new osd running node
+        6. Health check
+        """
+
+        # OSDs before running disk replacement
+        old_osds = []
+        for osd in pod.get_osd_pods():
+            old_osds.append(osd.name)
+        log.info(f"OSDs before disk replacement {old_osds}")
+
+        # Get osd running nodes
+        osd_nodes = node.get_osd_running_nodes()
+        log.info(f"osd_nodes {osd_nodes}")
+
+        # Fail and replace the osd device
```

Comment:
As per description of the bug, the customer deleted the entire VM including the Data disk of the node. Does the code here do the same for inducing the failure? If not, please implement.

Reply:
Closing this PR and opening a new one for VM deletion and replacement